src/mon/OSDMonitor.cc

```diff
@@ -1425,7 +1425,7 @@ bool OSDMonitor::preprocess_failure(MonOpRequestRef op)
     int from = m->get_orig_source().num();
     if (!osdmap.exists(from) ||
 	osdmap.get_addr(from) != m->get_orig_source_inst().addr ||
-	osdmap.is_down(from)) {
+	(osdmap.is_down(from) && m->if_osd_failed())) {
```

Comment:
If I'm reading this correct it means a failure report cancellation should not be ignored, even if the osd is down. That makes sense, but can you break that into a separate commit please?